frontends/main/src/app/dashboard/my-lists/[id]/page.tsx

```diff
@@ -1,14 +1,15 @@
 import React from "react"
 import { UserListDetailsContent } from "@/app-pages/DashboardPage/UserListDetailsContent"
-import { PageParams } from "@/app/types"
 import invariant from "tiny-invariant"
+import { PageParams } from "@/app/types"
 
-const Page: React.FC<PageParams<object, { id: number }>> = async ({
+const Page: React.FC<PageParams<never, { id: string }>> = async ({
   params,
 }) => {
-  const resolved = await params
-  invariant(resolved?.id, "id is required")
-  return <UserListDetailsContent userListId={resolved.id} />
+  const resolved = await params!
+  const id = Number(resolved.id)
+  invariant(!id, "id is required")
+  return <UserListDetailsContent userListId={id} />
```

Comment:
What was happening was... the list data was being fetched based on the URL parameter, which was a string. But when we invalidated the query cache, we were using a number. So:

```ts
// query stored under key
["userlist", "detail", "16", "items", listingParams]
// invalidated query used a number for the id:
["userlist", "detail", 16]
```
So we weren't invalidating the correct query.

---------

Manual typings are error prone... Putting `{ id: number }` as the page parameter types was reasonable, but it really shouldve been `{ id: string }`, in which case Typescript wouldve caught the issue.

NextJS 15.5 largely removes the need for manual typing of page params. See #2500